### PR TITLE
Try to prevent bugs by introducing Cppcheck

### PR DESCRIPTION
**Pull request by visuve**
- Stop only on error and warning level issues for now
  -  The ``--enable=warning`` contains errors too (tested)

### CODE REVIEW

**Comment by visuve** [comment]
The Codacy does not seem to pick a lot. If you find this useful, please merge.

File: parser/pe/DOSExe.cpp
Comment (hasherezade) [inline parser/pe/DOSExe.cpp:1]: @visuve - TBH, this is the change I don't like. I would rather have a single return at the end, as it was before.

**Comment by hasherezade** [comment]
Thank you @visuve! Except one change that I commented on, everything looks good.

File: parser/pe/DOSExe.cpp
Comment (visuve) [reply parser/pe/DOSExe.cpp:1]: I put it back. 

For me, I just do not like seeing empty exception handlers. I regard it as an anti-pattern 😅 But maybe in the future these exceptions could be logged e.g. with ``qDebug`` ?

Comment (hasherezade) [reply parser/pe/DOSExe.cpp:1]: I understand, and you have the point. But to me, a function that does not have an explicit return at the end, only conditional ones, is much worse.
I guess I could log an error here, but it is not that important. For now let's leave it as is.

**Pull request merged**